talisker/postgresql.py

```diff
@@ -56,11 +56,16 @@ def prettify_sql(sql):
 
 def get_safe_connection_string(conn):
     try:
-        # 2.7+
-        params = conn.get_dsn_parameters()
-    except AttributeError:
-        params = dict(i.split('=') for i in shlex.split(conn.dsn))
-    return '{user}@{host}:{port}/{dbname}'.format(**params)
+        try:
+            # 2.7+
+            params = conn.get_dsn_parameters()
+        except AttributeError:
+            params = dict(i.split('=') for i in shlex.split(conn.dsn))
+
+        params.setdefault('host', 'localhost')
+        return '{user}@{host}:{port}/{dbname}'.format(**params)
+    except Exception:
+        return None
```

Comment:
Should this be something more helpful other than None? Maybe 'Unable to parse'?

Reply:
That's what I had originally. I supposed it surfaces the information better.